iroh-relay/src/client.rs

```diff
@@ -37,6 +36,64 @@ pub(crate) mod streams;
 #[cfg(not(wasm_browser))]
 mod util;
 
+/// Connection errors
+#[allow(missing_docs)]
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum ConnectError {
+    #[error("Invalid URL for websocket {0}")]
+    InvalidWebsocketUrl(Url),
+    #[error("Invalid relay URL {0}")]
+    InvalidRelayUrl(Url),
+    #[error(transparent)]
+    Websocket(#[from] tokio_tungstenite_wasm::Error),
```

Comment:
I'm not sure this error is self-describing.  Maybe an `#[error("Websocket transport error: {0}")]` would be better?  (If that is the correct interpretation)

Which immediately makes me wonder "but what if I'm not using websocket"?  Why is there no generic IO or Transport error this falls under?